src/main/java/duke/tool/Parser.java

```diff
@@ -0,0 +1,110 @@
+                    return new Exit();
+                } else if (ValidInput.getCmdType(command[0]).name().equals("LIST")) {
+                    return new List();
+                }else if (ValidInput.getCmdType(command[0]).name().equals("DONE")) {
+                    if (command.length == 1) {
+                        throw new NoTaskChosenException(command[0]);
+                    } else if (!isInteger(command[1])
+                            || Integer.parseInt(command[1]) > numOfOrders) {
+                        throw new NoThisNumOfTaskException();
+                    } else {
+                        return new Done(Integer.parseInt(command[1]) - 1);
+                    }
+                } else if (ValidInput.getCmdType(command[0]).name().equals("DELETE")) {
+                    if (command.length == 1) {
+                        throw new NoTaskChosenException(command[0]);
+                    } else if (!isInteger(command[1])
+                            || Integer.parseInt(command[1]) > numOfOrders) {
+                        throw new NoThisNumOfTaskException();
+                    } else {
+                        return new Delete((Integer.parseInt(command[1]) - 1));
+                    }
+                } else if (ValidInput.getCmdType(command[0]).name().equals("FIND")) {
+                    return new Find(command[1]);
+                } else {
+                    if (command.length == 1) {
+                        throw new NoDescriptionException(command[0]);
+                    } else if (ValidInput.getCmdType(command[0]).name().equals("DEADLINE")) {
+                        String[] splitAgain = command[1].split("/by ");
+                        if (splitAgain.length == 1) {
+                            throw new NoTimeException(command[0]);
+                        }
+                        return new Deadline(splitAgain[0],
+                                LocalDateTime.parse(splitAgain[1], VALID_FORMAT), false);
+                    } else if (ValidInput.getCmdType(command[0]).name().equals("EVENT")) {
+                        String[] splitAgain = command[1].split("/at ");
+                        if (splitAgain.length == 1) {
+                            throw new NoTimeException(command[0]);
+                        }
+                        return new Event(splitAgain[0],
+                                LocalDateTime.parse(splitAgain[1], VALID_FORMAT), false);
+                    } else if (ValidInput.getCmdType(command[0]).name().equals("TODO")) {
+                        return new Todo(command[1], false);
+                    } else {
+                        throw new NoSuchOrderException();
+                    }
+                }
+            }
+        } catch (NoDescriptionException | NoTimeException | NoSuchOrderException
+                | NoTaskChosenException | NoThisNumOfTaskException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
```

Comment:
Perhaps, you might want to consider using a switch-case instead of nested if-else loops so that the code would look neater